keps/sig-windows/3636-windows-csi-host-process-pods/README.md

```diff
@@ -0,0 +1,1235 @@
+
+Pros:
+
+* Like implementation idea 2, we switch to a single notion of version via Go modules.
+* The pure function paradigm more accurately reflects the nature of the new design, which simplifies how clients use the
+  library.
+* Like implementation idea 2, new APIs are easier to add by moving away from code generation.
+
+Cons:
+
+* There is now an implicit dependency on the os API package-level variable. Testing can still be done by subbing out the
+  variable with a mock implementation during test time.
+* More work (2 imports -> 1, remove server initialization, replace function call and request type package names) needs
+  to be done by clients to adapt to the new change, though it’s not that much more than implementation idea 2. Again,
+  the price is only paid once.
+* Like impl idea 2, we also need to transition our API doc generation to generate from Go source.
+
+
+### Comparison Matrix
+
+| |Idea 1: Update the conversion layer to use the server code gRPC|Idea 2: Update the CSI Drivers to use the server code directly (preferred)|Idea 3: Convert CSI Proxy to a Library of Functions|
+| --- |--- |--- |--- |
+| Adoption cost|Minimal (only changing imports)|Considerate (imports and API calls)|Considerate (imports, API calls, and initialization)|
+| Future development|Still need code generation and and protobuf|Directly add methods to Go code, but leaves legacy notion of “server”|Directly add functions to Go code. Code base cleaned up|
+| Versioning|Both Go mod version and API version are maintained|Go mod version only|Go mod version only|
+| Testing|Current tests should still work.|Current tests should still work.|OS API mocking needs to be subbed in, as we have an implicit dependency|
+| Support for legacy client/server model|Still supported|Not supported|Not supported|
+
+
+### Maintenance of the new model and existing client/server model of CSI Proxy
+
+The `library-development` branch will be used for the development of this model. `master` will have the existing
+client/server mode. We plan to create alpha tags on the `library-development` branch and use it in CSI Drivers.
+Once integrated, we will create a v2 tag and make `library-deveopment` the new default. `master` will point
+to the new implementation, whereas the legacy code is maintained on the`v1.x`.
+
+`v1.x` will still be open for urgent bug fixes but new features should be developed in the v2 codebase.
+
+### Security analysis
+
+We suggest enabling HostProcess containers only for system workloads. One of the drawbacks of this new mode is the
+lack of filesystem isolation, which means that HostProcess containers have entire visibility of the host fileystem.
+To prevent workloads from accessing specific paths in the host we recommend:
+
+- Creating a Windows user with limited permissions to create files under the kubelet controlled path `C:\var\lib\kubelet`
+- Enabling `PodSecurityAdmission` on the namespace where system components run. Note that this needs an additiona
```

Comment:
I'm not sure this will buy someone anything as the CSI plugin would have to run as "Privileged" in order to have .windowsOptions.hostProcess=true. So it's really just opening up the namespace where it runs.

Reply:
I read more about Pod Security Standards and how it works with the Pod Security Admission controller, I've updated this part with a section for Cluster Administrators, please take a look again.